src/main/java/org/reactome/addlinks/fileprocessors/ensembl/EnsemblBiomartFileProcessor.java

```diff
@@ -0,0 +1,115 @@
+package org.reactome.addlinks.fileprocessors.ensembl;
+
+import org.reactome.addlinks.fileprocessors.FileProcessor;
+
+import java.io.*;
+import java.util.*;
+
+public class EnsemblBiomartFileProcessor extends FileProcessor<Map<String, List<String>>> {
+
+    public EnsemblBiomartFileProcessor()
+    {
+        super(null);
+    }
+
+    public EnsemblBiomartFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    // TODO: Add unit tests
+    /**
+     * This class builds a large, rather unwieldy two-layer mapping structure of identifiers corresponding to a specific species.
+     * The first level has 4 types of keys for each species. These keys are 'uniprotToProteins', 'proteinToGenes',
+     * 'proteinToTranscripts', and 'transcriptToMicroarrays'. These describe the type of mapping that this
+     * key is to. For example, the key 'hsapiens_uniprotToProteins' would access a Map of Human UniProt protein identifiers
+     * to a List of Ensembl protein identifiers. Transcript denotes Ensembl transcript identifiers, Gene is Ensembl gene identifiers
+     * and Microarray denotes microarray probe identifiers.
+     * @return Map<String, Map<String, List<String>>>> mappings, the two-layer mapping structure of species-(identifier-[identifiers]).
+     * @throws IOException Can be thrown if file does not exist.
+     */
+    @Override
+    public Map<String, Map<String, List<String>>> getIdMappingsFromFile() throws IOException {
+
+        Map<String, Map<String, List<String>>> mappings = new HashMap<>();
+        // Iterate through each biomart file in the addlinks directory
+        File biomartDir = new File(this.pathToFile.toString());
+        for (File biomartFile : biomartDir.listFiles()) {
```

Comment:
IntelliJ mentioned that the `listFiles()` method could return a null leading to an NPE here.  An alternative for looping over files in a directory with the java.nio.Files class is:

```
try (Stream<Path> paths = Files.walk(this.pathToFile)) {
    paths
        .filter(Files::isRegularFile)
        .forEach(bioMartFile -> {
            ...
        });
}
```

Reply:
Done